tools/antithesis-go-generator/catalog_output.go

```diff
@@ -0,0 +1,162 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strings"
+	"text/template"
+)
+
+type GenInfo struct {
+	ExpectedVals      []*AntExpect
+	ExpectPackageName string
+}
+
+const GENERATED_SUFFIX = "_antithesis.go"
+
+func expect_output_file(module_name string) (*os.File, error) {
+	from_file := os.Getenv("GOFILE")
+	if from_file == "" {
+		from_file = module_name
+	}
+	dir_name, file_name := path.Split(from_file)
+	ext := path.Ext(file_name)
+	if len(ext) > 0 {
+		file_name = strings.TrimSuffix(file_name, ext)
+	}
+	generated_name := fmt.Sprintf("%s%s", file_name, GENERATED_SUFFIX)
+	output_file_name := path.Join(dir_name, generated_name)
+	fmt.Printf("Generated file: %q\n", output_file_name)
+
+	var file *os.File
+	var err error
+	if file, err = os.OpenFile(output_file_name, os.O_RDWR|os.O_CREATE, 0644); err != nil {
+		file = nil
+	}
+	if file != nil {
+		if err = file.Truncate(0); err != nil {
+			file = nil
+		}
+	}
+	return file, err
+}
+
+func assertion_name_repr(s string) string {
+	if s == "Reachable" || s == "Unreachable" {
```

Comment:
It feels like somewhere we should have a "string enum" of these values. Something like:

```
enum AssertionType {
    REACHABLE = "Reachable",
    UNREACHABLE = "Unreachable",
...
```

And then not have random strings in the code.

And possibly have more enums for the other magic strings like "must_be_hit" and "universal_test".

Reply:
Holding off on this now.  It should be revisited when we add assertion types like: `Perhaps`, `Eventually`, `Consistently` and `SometimesEach`, and we have a more complete taxonomy of these function signatures